README.md

```diff
@@ -28,4 +28,47 @@ You must have [cekit](https://github.com/cekit/cekit) installed.
 ## Building the JDK21 images
 
 * `cd wildfly-builder-image; cekit build --overrides=jdk21-overrides.yaml docker`
-* `cd wildfly-runtime-image; cekit build --overrides=jdk21-overrides.yaml docker`
\ No newline at end of file
+* `cd wildfly-runtime-image; cekit build --overrides=jdk21-overrides.yaml docker`
+
+## Tests executed for new Pull request
+
+###  Steps
+
+Behave tests are run when a PR is opened against the wildfly-s2i repository.
+The github action workflow [file](https://github.com/wildfly/wildfly-s2i/blob/main/.github/workflows/main.yml) is run.
+The steps are as follow:
+* Build multiarch images for all supported JDK.
+* Then, for each behave feature [files](https://github.com/wildfly/wildfly-s2i/tree/main/wildfly-builder-image/tests/features):
+* Execute the test, and redirect logs to `<feature name>-ubuntu-latest-${{ matrix.jdk }}.txt`
+* call `docker system prune` to release unreferenced images.
+* If a failure occurs, the workflow is aborted and the logs are collected.
+
+NOTE: Some tests are not supported on JDK11 (for example, starting WildFly 32, WildFly preview requires JDK17 as its minimal JDK version). Such tests 
+are not executed on JDK11 and are all located in the feature file: `no-jdk11.feature`
+
+### Understanding the failure logs
+
+The features test execution has been split into multiple executions due to created images resource consumption and log size.
+Each feature file has its own execution with its own log.
+When a failure occurs:
+* Check the Github action execution log. Note the feature file name that failed.
+* Download the log files.
+* Unzip the log files and search for the `wildfly-s2i-test-logs-ubuntu-latest-<jdk>/**/test-logs-legacy-s2i-ubuntu-latest-jdk11.txt<feature name>-ubuntu-latest-<jdk>.txt` file.
+* Access the end of the file: `tail -f --lines 100 <path of the file>`
+* You should see something like:
+```
+Failing scenarios:
```

Comment:
Can presence of failing scenarios be indicated in Github UI, so one know he should dig deeper what is going on?

Reply:
Good point. I added it. BTW, I added an extra transient commit to make the first executed test to fail.